src/components/table/base-table/base-table.js

```diff
@@ -329,51 +339,81 @@ export default class BaseTable extends React.Component {
+    if (renderParams.rowIndex === 0) {
+      return (
+        <ClassNames>
+          {({ css: makeClassName }) => (
+            <div
+              style={{
+                ...(renderParams.style || {}),
+                ...(columnDefinition.headerStyle || {}),
+              }}
+              className={classnames(
+                makeClassName({
+                  background: vars.colorNavy,
+                  color: vars.colorWhite,
+                }),
+                columnDefinition.headerClassName,
+                makeClassName({ textAlign: columnDefinition.align || 'left' })
+              )}
+            >
+              {this.headerRenderer(columnDefinition)}
+            </div>
+          )}
+        </ClassNames>
+      );
+    }
     return (
-      <CellMeasurer
-        cache={this.cellMeasurerCache}
-        columnIndex={renderParams.columnIndex}
-        key={renderParams.key}
-        parent={renderParams.parent}
-        rowIndex={renderParams.rowIndex}
-      >
-        {renderParams.rowIndex === 0 ? (
-          <div
-            style={renderParams.style}
-            className={classnames(
-              styles['header-cell'],
-              columnDefinition.headerClassName,
-              styles[`align-${columnDefinition.align || 'left'}`]
-            )}
-          >
-            {this.headerRenderer(columnDefinition)}
-          </div>
-        ) : (
+      <ClassNames>
+        {({ css: makeClassName }) => (
           <div
-            // TODO: move the className and handlers into Cell
-            style={renderParams.style}
             className={classnames(
-              styles[`align-${columnDefinition.align || 'left'}`],
-              {
-                [styles['hovered-row']]:
-                  renderParams.rowIndex === this.state.hoveredRowIndex &&
-                  renderParams.rowIndex !== 0 &&
-                  this.props.onRowClick,
-              },
+              makeClassName({
+                textAlign: columnDefinition.align || 'left',
+                ...(renderParams.rowIndex === this.state.hoveredRowIndex &&
+                renderParams.rowIndex !== 0 &&
+                this.props.onRowClick
+                  ? {
+                      cursor: 'pointer',
+                      transition: vars.transitionStandard,
+                      background: vars.colorGray90,
+                    }
+                  : {}),
+              }),
               columnDefinition.className,
+              columnDefinition.cellClassName,
               columnDefinition.classNameGetter
                 ? columnDefinition.classNameGetter({
                     rowIndex: this.getBodyRowIndex(renderParams.rowIndex),
                     columnKey: columnDefinition.key,
                     height: renderParams.height,
                     width: renderParams.width,
                   })
+                : undefined,
+              columnDefinition.cellClassNameGetter
```

Comment:
Just suggestions:

1. Defaulting it could avoid the check
2. `getCellClassName` Maybe? Then the `*Getter` is not rly needed

Reply:
Hmm, I wouldn't change the name, otherwise it's a breaking change as this is already used (actually the `classNameGetter` is, the `cellClassNameGetter` is just an alias, but they should have a consistent name).

Reply:
Ah didn't notice that. Makes total sense. Sorry. Yay for non-breaking changes.